File: packages/csharp/src/Errors/EndpointRemovedException.cs
namespace Tesote.Sdk.Errors;

/// <summary>
/// Raised when calling a method whose upstream endpoint is gone in this API
/// version. The SDK keeps the method per the back-compat policy but throws this
/// pointing at the replacement.
/// </summary>
public sealed class EndpointRemovedException : TesoteException
{
    /// <summary>Construct with a human-readable message.</summary>
    public EndpointRemovedException(string message)
        : base(message, "ENDPOINT_REMOVED", 0, null, null, null, null, null, 0, null)
    {
    }
}

File: packages/php/src/Errors/EndpointRemovedException.php
<?php

declare(strict_types=1);

namespace Tesote\Sdk\Errors;

use RuntimeException;

/**
 * Thrown when a method's upstream endpoint has been removed in the API
 * version the caller targeted. Carries a hint at the replacement.
 */
final class EndpointRemovedException extends RuntimeException
{
    public function __construct(string $method, public readonly ?string $replacement = null)
    {
        $msg = sprintf('Endpoint %s has been removed.', $method);
        if ($replacement !== null) {
            $msg .= ' Use ' . $replacement . ' instead.';
        }
        parent::__construct($msg);
    }
}
